src/main/java/org/jenkinsci/plugins/codesonar/services/MaxCyclomaticComplexityJsonReader.java

```diff
@@ -0,0 +1,296 @@
+        switch(token) {
+        case NAME:
+            String name = jsonReader.nextName();
+            LOGGER.fine(NAME_TOKEN + name);
+            /*
+             * If it has specified a special action to take based off of
+             * the name parameter, than call it.
+             */
+            if(action != null) {
+                action.take(name, parameter);
+            }
+            break;
+        case STRING:
+            LOGGER.fine(STRING_TOKEN + jsonReader.nextString());
+            break;
+        case NUMBER:
+            LOGGER.fine(NUMBER_TOKEN + jsonReader.nextLong());
+            break;
+        case BOOLEAN:
+            LOGGER.fine(BOOLEAN_TOKEN + jsonReader.nextBoolean());
+            break;
+        case NULL:
+            jsonReader.nextNull();
+            LOGGER.fine(NULL_TOKEN);
+            break;
+        default:
+            LOGGER.fine(MessageFormat.format("Unexpected token type: \"{0}\"", token.name()));
+        }
+        
+    }
+
+    public ProcedureMetric readProcedureMetric() throws IOException {
+        JsonReader jsonReader = new JsonReader(inputStreamReader);
+        jsonReader.setLenient(true);
+
+        try {
+            while (jsonReader.hasNext()) {
+                JsonToken nextToken = jsonReader.peek();
+                
+                if (JsonToken.BEGIN_OBJECT.equals(nextToken)) {
+
+                    jsonReader.beginObject();
+
+                } else if (JsonToken.END_OBJECT.equals(nextToken)) {
+
+                    jsonReader.endObject();
+
+                } else {
+                    parseToken(jsonReader.peek(), jsonReader, new SimpleParseAction() {
+                        @Override
+                        public void take(String name) throws IOException {
+                            if(BUILD_LAUNCHD_OBJECT.equals(name)) {
+                                parseBuildLaunchd(jsonReader);
+                            } else if(LAUNCHD_OBJECT.equals(name)) {
+                                parseLaunchd(jsonReader);
+                            } else if(PROJECT_OBJECT.equals(name)) {
+                                parseProject(jsonReader);
+                            } else if(ROWS_ARRAY.equals(name)) {
+                                List<ProcedureMetric> rowsArray = parseRowsArray(jsonReader);
+                                // Sort rows array by "cyclomatic complexity" firstly and by "procedure" secondly
+                                LOGGER.fine("rowsArray (before sorting)=" + rowsArray);
+                                Collections.sort(rowsArray);
+                                LOGGER.fine("rowsArray (after sorting)=" + rowsArray);
+                                /*
+                                 *  Get the first element of the list, which according to sorting settings,
+                                 *  corresponds to the one with the maximum cyclomatic complexity. 
+                                 */

```

Comment:
This code should mention that it is making some assumptions compatible with the ProcedureService.  I suggest:

```
// Read all rows into an array so that we can sort them.
// We will pick the first item that matches the sorting scheme implemented on the `ProcedureMetric` type.
// This is done to be compatible with assumptions made by `ProceduresService.getProcedureWithMaxCyclomaticComplexity()`.
```